pkg/capabilities/cli/cmd/generate_types.go

```diff
@@ -0,0 +1,219 @@
+			},
+		})
+	},
+}
+
+func GenerateTypes(dir string, helpers []WorkflowHelperGenerator) error {
+	schemaPaths, err := schemaFilesFromDir(dir)
+	if err != nil {
+		return err
+	}
+
+	cfgInfo, err := ConfigFromSchemas(schemaPaths)
+	if err != nil {
+		return err
+	}
+
+	for _, schemaPath := range schemaPaths {
+		if err = generateFromSchema(schemaPath, cfgInfo, helpers); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func generateFromSchema(schemaPath string, cfgInfo ConfigInfo, helpers []WorkflowHelperGenerator) error {
+	allFiles := map[string]string{}
+	file, content, err := TypesFromJSONSchema(schemaPath, cfgInfo)
+	if err != nil {
+		return err
+	}
+
+	capabilityType := cfgInfo.SchemaToTypeInfo[schemaPath].CapabilityType
+	if err = capabilityType.IsValid(); err != nil {
+		return fmt.Errorf("invalid capability type %v", capabilityType)
+	}
+
+	allFiles[file] = content
+	typeInfo := cfgInfo.SchemaToTypeInfo[schemaPath]
+	structs, err := generatedInfoFromSrc(content, getCapID(typeInfo), typeInfo)
+	if err != nil {
+		return err
+	}
+
+	if err = generateHelpers(helpers, structs, allFiles); err != nil {
+		return err
+	}
+
+	if err = printFiles(path.Dir(schemaPath), allFiles); err != nil {
+		return err
+	}
+
+	fmt.Println("Generated types for", schemaPath)
+	return nil
+}
+
+func getCapID(typeInfo TypeInfo) *string {
+	id := typeInfo.SchemaID
+	idParts := strings.Split(id, "/")
+	id = idParts[len(idParts)-1]
+	var capID *string
+	if strings.Contains(id, "@") {
+		capID = &id
+	}
+	return capID
+}
+
+func schemaFilesFromDir(dir string) ([]string, error) {
+	var schemaPaths []string
+
+	if err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Ignore directories and files that don't match the CapabilitySchemaFileExtension
+		if info.IsDir() || !CapabilitySchemaFilePattern.MatchString(path) {
+			return nil
+		}
+
+		schemaPaths = append(schemaPaths, path)
+		return nil
+	}); err != nil {
+		return nil, fmt.Errorf("error walking the directory %v: %v", dir, err)
+	}
+	return schemaPaths, nil
+}
+
+func generateHelpers(helpers []WorkflowHelperGenerator, structs GeneratedInfo, allFiles map[string]string) error {
+	for _, helper := range helpers {
+		files, err := helper.Generate(structs)
+		if err != nil {
+			return err
+		}
+
+		for f, c := range files {
+			if _, ok := allFiles[f]; ok {
+				return fmt.Errorf("file %v is being created by more than one generator", f)
+			}
+			allFiles[f] = c
+		}
+	}
+	return nil
+}
+
+func ConfigFromSchemas(schemaFilePaths []string) (ConfigInfo, error) {
+	configInfo := ConfigInfo{
+		Config: generator.Config{
+			Tags:   []string{"json", "yaml", "mapstructure"},
+			Warner: func(message string) { fmt.Printf("Warning: %s\n", message) },
+		},
+		SchemaToTypeInfo: map[string]TypeInfo{},
+	}
+
+	for _, schemaFilePath := range schemaFilePaths {
+		jsonSchema, err := schemas.FromJSONFile(schemaFilePath)
```

Comment:
validate the schemaFilePath against the regex first and return clear error msg. iiuc, failure will happen on 176, but user won't know what the pattern is supposed to be